sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/CoverageTest.java

```diff
@@ -202,16 +205,68 @@ public void givenMultipleBranchPointsPerLine_getBranchCoverage_returnsBranchCove
   public void givenMultipleBranchPointsPerLineInDifferentFiles_getBranchCoverage_returnsBranchCoverage() {
     final String firstPath = "firstPath";
     final String secondPath = "secondPath";
+    final String coverageIdentifier = "coverageIdentifier";
 
     Coverage sut = new Coverage();
-    sut.add(new BranchPoint(firstPath, 1, 1, 3, 0, 2));
-    sut.add(new BranchPoint(firstPath, 1, 4, 6, 1, 1));
+    sut.add(new BranchPoint(firstPath, 1, 1, 3, 0, 2, coverageIdentifier));
+    sut.add(new BranchPoint(firstPath, 1, 4, 6, 1, 1, coverageIdentifier));
 
-    sut.add(new BranchPoint(secondPath, 1, 5, 8, 0, 2));
-    sut.add(new BranchPoint(secondPath, 1, 10, 12, 1, 0));
-    sut.add(new BranchPoint(secondPath, 1, 12, 14, 2, 0));
+    sut.add(new BranchPoint(secondPath, 1, 5, 8, 0, 2, coverageIdentifier));
+    sut.add(new BranchPoint(secondPath, 1, 10, 12, 1, 0, coverageIdentifier));
+    sut.add(new BranchPoint(secondPath, 1, 12, 14, 2, 0, coverageIdentifier));
 
-    assertThat(sut.getBranchCoverage(firstPath)).containsOnly(new BranchCoverage(1, 2, 2));
-    assertThat(sut.getBranchCoverage(secondPath)).containsOnly(new BranchCoverage(1, 3, 1));
+    assertThat(sut.getBranchCoverage(firstPath)).containsExactlyInAnyOrder(new BranchCoverage(1, 2, 2));
+    assertThat(sut.getBranchCoverage(secondPath)).containsExactlyInAnyOrder(new BranchCoverage(1, 3, 1));
+  }
+
+  @Test
+  public void givenBranchPointsFromFromDifferentReports_getBranchCoverage_returnsBranchCoverage(){
```

Comment:
We don't have tests that cover real scenarios, just simulated ones. Are you confident enough that you'll manage to do a proper validation? You can try retrieving coverage data of some files/projects that might be affected and see how they are behaving with the new computation